fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
```

Comment:
I'm less talking about the specifics of YARPC or outbounds in general, but rather illustrating how the configuration can impact the number of components that the module has to return. I'm ok to YAGNI and punt on this for now, but I feel pretty strongly that configuration is a special thing that should be used as part of the module creation process, rather than a dependency of a component.